dev-docs/source/Testing/SliceViewer/SliceViewer.rst

```diff
@@ -13,226 +13,412 @@ The Sliceviewer in Workbench has the joint functionality of the SpectrumViewer a
+
+3. Double click a row
+
+    - It should change the slicepoint along the integrated momentum axis and zoom into the peak - e.g. in (X,Y) = (H,K) then the slicepoint along L will be set to 1 and there will be a cross at (0,0).
+    - Note for ``md_4D`` the cross should be plotted at all E (obviously a Bragg peak will only be on the elastic line but the peak object has no elastic/inelastic logic and the sliceviewer only knows that `E` is not a momentum axis, it could be temperature etc.).
+
+4. Click Add Peaks in the Peak Actions section at the top of the peak viewer
+5. Click somewhere in the colorfill plot
+
+    - Confirm a peak has been added to the table at the position you clicked
+
+6. Click Remove Peaks
+7. Click on the cross corresponding to the peak you just added
+
+    - Confirm the correct row has been removed from the table
+    - The cross should be removed from the plot
+
+8. Repeat the above steps 1-7 in non-orthogonal view.
+
+MDHistoWorkspace
+~~~~~~~~~~~~~~~~
+
+1. Make a 2D and 3D MDHistoWorkspaces
 
 .. code-block:: python
 
-    ws = CreateMDWorkspace(Dimensions='3', Extents='-3,3,-3,3,-3,3',
-                       Names='H,K,L', Units='r.l.u.,r.l.u.,r.l.u.',
-                       Frames='HKL,HKL,HKL',
-                       SplitInto='2', SplitThreshold='10')
-    expt_info = CreateSampleWorkspace()
-    ws.addExperimentInfo(expt_info)
-    SetUB(ws, 1,1,2,90,90,120)
-    BinMD(InputWorkspace=ws, AxisAligned=False,
-        BasisVector0='[00L],r.l.u.,0,0,1',
-        BasisVector1='[HH0],r.l.u.,1,1,0',
-        BasisVector2='[-HH0],r.l.u.,-1,1,0',
-        OutputExtents='-4,4,-4,4,-0.25,0.25',
-        OutputBins='101,101,1', OutputWorkspace='ws_slice', NormalizeBasisVectors=False)
+    md_3D_histo = BinMD(InputWorkspace='md_4D', AlignedDim0='H,-2,2,100', AlignedDim1='K,-1,1,100', AlignedDim2='L,-1.5,1.5,100')
+
+2. Open ``md_3D_histo`` in sliceviewer it should not support dynamic rebinning (can't change number of bins).
+3. Test the toolbar buttons pan, zoom, line plots, ROI as in step 3 of the :ref:`toolbar-checklist` instructions.
+4. Test changing/swapping viewing axes
+5. Test the :ref`Nonorthogonal view<mantid:sliceviewer_nonortho>` as above
+6. Open ``md_4D_svrebinned`` in sliceviewer (should be in the ADS after preceding tests).
+
+    - It should support dynamic rebinning (i.e. will be able to change number of bins along each axis).
 
-- Run the above code and open ``ws_slice`` in the sliceviewer.
-- Check that non-orthogonal view is enabled on opening - however when clicked the gridlines are orthogonal  (in this case 110 is orthogonal to 001).
+7. With ``md_4D_svrebinned`` open in the sliceviewer, delete ``md_4D`` in the ADS.
 
-C. Check that changing the properties of the workspace that governs the support for the non-orthogonal transform closes the sliceviewer window.
+    - It should close sliceviewer because the support for dynamic rebinning has changed
```

Comment:
It didn't close, but dynamic rebinning seems fine... I think the rebinning seemed fine because it had produced an ``md_4d_svrebinned_sv_rebinned`` workspace. But it doesn't close, even if I open ``md_4D_svrebinned``, do not zoom and delete md_4D

Reply:
Ah yes that's related to a memory leak - we know about that but I can't find the issue...

Reply:
Have made a new issue for this #33434